src/loan_form/LoanForm.jsx

```diff
@@ -65,27 +65,28 @@ export default class LoanForm extends React.Component {
     return 30.417
   }
 
-  effectiveMonthlyInterest(amountOwed, yearlyInterestRate, month, year) {
-    return amountOwed * (((1 + yearlyInterestRate / this.daysInYear(year)) ** this.daysInMonth(month)) - 1)
+  effectiveMonthlyInterestRate(yearlyInterestRate, month, year) {
+    return ((1 + yearlyInterestRate / this.daysInYear(year)) ** this.daysInMonth(month)) - 1
   }
 
   calculateRepaymentTerm({debt:{amountOwed, interestRate: {rate}}, paymentPlan: {monthlyPayment}}) {
+    const a = Math.log(monthlyPayment),
+          b = Math.log(monthlyPayment - amountOwed * this.effectiveMonthlyInterestRate(rate)),
+          c = Math.log(1 + this.effectiveMonthlyInterestRate(rate))
     
-    let tempAmountOwed = parseFloat(amountOwed)
-    let interest = this.effectiveMonthlyInterest(tempAmountOwed, rate)
-    let repaymentTerm = 0
+    const repaymentTerm = rate ? (a - b) / c : amountOwed / monthlyPayment
```

Comment:
hmm it really seems like there should be a better name for what `a`,`b`, and`c` are given that the formula using them looks so much like `amountOwed / monthlyPayment`

Reply:
They're only there in an attempt to make the single equation more readable, they're otherwise arbitrary segments